File: frontend/youthconnect-frontend/src/app/services/event.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface RsvpRequest {
    eventId: number;
    userId: number;
}

export interface EventResponse {
    eventId: number;
    title: string;
    description: string;
    eventDate: string;
    location: string;
    createdByAdminId: number;
    status: string;
    createdAt: string;
    updatedAt?: string;
    rsvpCount: number;
    expectedCount: number;
}

export interface AttendanceResponse {
    attendanceId: number;
    eventId: number;
    userId: number;
    isAttended: boolean;
    registeredAt: string;
    attendedAt?: string;
}

export interface EventRequest {
    title: string;
    description: string;
    eventDate: string;
    location: string;
    createdByAdminId: number;
    status?: string;
}

@Injectable({
    providedIn: 'root'
})
export class EventService {
    private http = inject(HttpClient);
    private apiUrl = 'http://localhost:8080/api/events';
    private skApiUrl = 'http://localhost:8080/api/sk/events';

    // SK Official endpoints
    createEvent(request: EventRequest): Observable<EventResponse> {
        return this.http.post<EventResponse>(this.skApiUrl, request);
    }

    editEvent(eventId: number, request: EventRequest): Observable<EventResponse> {
        return this.http.put<EventResponse>(`${this.skApiUrl}/${eventId}`, request);
    }

    deleteEvent(eventId: number): Observable<any> {
        return this.http.delete(`${this.skApiUrl}/${eventId}`, { responseType: 'text' });
    }

    getAllEvents(): Observable<EventResponse[]> {
        return this.http.get<EventResponse[]>(this.skApiUrl);
    }

    // Youth endpoints (existing)
    getEventById(eventId: number): Observable<EventResponse> {
        return this.http.get<EventResponse>(`${this.apiUrl}/${eventId}`);
    }

    rsvpEvent(request: RsvpRequest): Observable<any> {
        return this.http.post(`${this.apiUrl}/rsvp`, request, { responseType: 'text' });
    }

    cancelRsvp(eventId: number, userId: number): Observable<any> {
        return this.http.delete(`${this.apiUrl}/${eventId}/rsvp/${userId}`, { responseType: 'text' });
    }

    getOwnRsvps(userId: number): Observable<AttendanceResponse[]> {
        return this.http.get<AttendanceResponse[]>(`${this.apiUrl}/rsvp/user/${userId}`);
    }

    getEventRsvps(eventId: number): Observable<AttendanceResponse[]> {
        return this.http.get<AttendanceResponse[]>(`${this.skApiUrl}/${eventId}/rsvp`);
    }
}

File: backend/youthconnect_id/src/main/java/com/youthconnect/youthconnect_id/services/EventService.java
package com.youthconnect.youthconnect_id.services;

import java.util.List;

import com.youthconnect.youthconnect_id.dto.AttendanceResponse;
import com.youthconnect.youthconnect_id.dto.EventRequest;
import com.youthconnect.youthconnect_id.dto.EventResponse;
import com.youthconnect.youthconnect_id.dto.MarkAttendanceRequest;
import com.youthconnect.youthconnect_id.dto.RsvpRequest;

public interface EventService {
    EventResponse createEvent(EventRequest request);
    EventResponse editEvent(int eventId, EventRequest request);
    void deleteEvent(int eventId);
    List<EventResponse> getAllEvents();
    AttendanceResponse markAttendance(int eventId, MarkAttendanceRequest request);

    EventResponse getEventById(int eventId);
    AttendanceResponse rsvpEvent(RsvpRequest request);
    void cancelRsvp(int eventId, int userId);
    List<AttendanceResponse> getOwnRsvps(int userId);
    List<AttendanceResponse> getEventRsvps(int eventId);
}